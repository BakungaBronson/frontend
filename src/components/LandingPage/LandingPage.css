.LandingPageMain {
  height: 100vh;
  display: flex;
  flex-direction: column;
}

.LandingPageMainContent {
  flex: 1;
  display: flex;
  flex-direction: row;
  align-self: center;
  padding-left: 2rem;
  padding-right: 2rem;
}

.LandingPageMainContentInfo {
  display: grid;
  align-items: center;
}

.LandingPageMainContentImg {
  display: grid;
  align-items: center;
}

.LandingPageMainContentInfo h1 {
  font-size: 32px;
  color: #F7B21F;
}

.LandingPageMainContentInfo h3 {
  color: #ffffff;
  line-height: 37px;
  font-size: 32px;
  width: 80%;
}

@media only screen and (max-width: 640px) {
  .LandingPageMainContentImg {
    display: none;
  }
}