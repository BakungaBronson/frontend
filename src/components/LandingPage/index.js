import React from 'react';
import { Link, withRouter} from 'react-router-dom';
import { connect } from 'react-redux';
import PrimaryButton from '../PrimaryButton';
import Header from '../Header';
import LandingFooter from '../LandingFooter';
import { ReactComponent as InfrastrucurePic } from '../../assets/images/infrastructure.svg';
import './LandingPage.css';

const LandingPage = (props) => {
  let { user } = props;
  if (user.data.id === undefined) {
    user = false;
  }
  return (
    <div className="LandingPageMain">
      <Header />
      <div className="LandingPageMainContent">
        <div className="LandingPageMainContentInfo">
          <div>
            <h1 className="uppercase extra-bold">managed kubernetes platform</h1>
            <h3 className="bold">Automated container deployment, scaling and management.</h3>
            {
              user ? (
                <Link to={`/users/${user.data.id}/projects`}><PrimaryButton label="get started" /></Link>
              ) : (
                <Link to="/register"><PrimaryButton label="get started" /></Link>
              )
            }

          </div>
        </div>
        <div className="LandingPageMainContentImg">
          <InfrastrucurePic className="LandingPageInfrastructurePic" />
        </div>
      </div>
      <div className="LandingPageFooter">
        <LandingFooter />
      </div>
    </div>
  );
};


const mapStateToProps = (state) => {
  const { user } = state;
  return { user };
};


export default connect(mapStateToProps)(withRouter(LandingPage));
