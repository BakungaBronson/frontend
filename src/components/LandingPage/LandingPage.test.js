import React from 'react';

import { shallow } from 'enzyme';

import LandingPage from './';

const props = {
    user: {
        data: {
            id: ""
        }
    }
}
describe('LandingPage Component', () => { 
    const LandingPageComponent = shallow(<LandingPage.WrappedComponent {...props} />);
    it('Matches snapshot', () => {
        expect(LandingPageComponent).toMatchSnapshot();
    })
})
