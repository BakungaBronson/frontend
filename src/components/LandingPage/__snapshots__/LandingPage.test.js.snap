// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage Component Matches snapshot 1`] = `ShallowWrapper {}`;
